colossalai/zero/sharded_model/sharded_model_v2.py

```diff
@@ -79,7 +80,8 @@ def __init__(self,
         self._require_backward_grad_sync: bool = True
 
     def forward(self, *args: Any, **kwargs: Any) -> torch.Tensor:
-        args, kwargs = cast_float_arguments(cast_tensor_to_fp16, *args, **kwargs)
+        # TODO args can be Long!
```

Comment:
`cast_tensor_to_fp16` will check whether the tensor if float, if not it won't be cast to fp16.

Reply:
DONE, revert the origin code.